pkg/proxy/service/cluster_service.go

```diff
@@ -65,7 +65,49 @@ func (s *ServiceImpl) GetClusterAccess(userID, username, workspace, proxyPluginN
 		return nil, fmt.Errorf("the requested space is not available")
 	}
 
-	return s.accessForSpace(space, signup.CompliantUsername, proxyPluginName)
+	return s.accessForSpace(space, userName, proxyPluginName)
+}
+
+func (s *ServiceImpl) getUserSignupComplaintName(userID, username string, publicViewerEnabled bool) (string, error) {
+	// if PublicViewer is enabled and the requested user is the PublicViewer, than no lookup is required
+	if publicViewerEnabled && username == userID && userID == toolchainv1alpha1.KubesawAuthenticatedUsername {
+		return username, nil
+	}
+
+	// retrieve the UserSignup from cache
+	//
+	// don't check for usersignup complete status, since it might cause the proxy blocking the request
+	// and returning an error when quick transitions from ready to provisioning are happening.
+	userSignup, err := s.Services().SignupService().GetSignupFromInformer(nil, userID, username, false)
+	if err != nil {
+		return "", err
+	}
+	// if signup has the CompliantUsername set it means that MUR was created and useraccount is provisioned
+	if userSignup == nil || userSignup.CompliantUsername == "" {
+		cause := errs.New("user is not provisioned (yet)")
+		log.Error(nil, cause, fmt.Sprintf("signup object: %+v", userSignup))
+		return "", cause
+	}
```

Comment:
I see this check is now duplicated at lines 99-108. Maybe we could extract this into a function ?

Reply:
good catch. done in f9b9b5a0fd1166c3f38d05f4ce80006866522624